daemon/oci_linux.go

```diff
@@ -533,7 +533,7 @@ func setMounts(daemon *Daemon, s *specs.Spec, c *container.Container, mounts []c
 	//  - /dev/shm, in case IpcMode is none.
 	// While at it, also
 	//  - set size for /dev/shm from shmsize.
-	var defaultMounts []specs.Mount
+	defaultMounts := s.Mounts[:0]
```

Comment:
but now we're copying all mounts, but the code inside the loop skips certain mounts? (ie, old code is still there and also did something else)

Reply:
the idea of copying a slice in place is you skip certain items, which is exactly what the existing code does. In can be boiled down to:

```go
defaultMounts := s.Mounts[:0]
for _, m := range mounts {
   // filter out some mounts
   if weDoNotWantThisMount(m) {
      continue
   }
   defaultMounts = append(defaultMounts, m)
}
```

Basically what happens is both slices share the same storage (and have the same capacity). At the very beginning the size of the new slice is 0, and it grows by one at each `append()`. Internally, it's just copying pointers. The best thing about this approach is there's no need to allocate and copy the new elements. The side effect is the original slice gets screwed up, so it can't used after.

It is also described here: https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating, and this is a quick example to play with: https://play.golang.org/p/E8hnGayR2ht

Reply:
Oh! Nevermind, I was reading `s.Mounts[0:]`; yes, I recall seeing this trick a while back; that _does_ modify the original `s.Mounts` though, but looks like that's not an issue here, because we overwrite it later on any way (`s.Mounts = defaultMounts`)